### PR TITLE
issue #29788 - itemlocdist_source_id_idx - 15min to 1.5min

### PR DESCRIPTION
**Pull request by gilmoskowitz**
Adding an index on the source id improves performance significantly.
Adding a foreign key constraint on itemlocdist_itemlocdist_id_fkey
should also help but exposes a bug in inventory distribution
cleanup.
Simplifying the formatlocationname function speeds it up by
about 10% (not much but the code is much easier to read now).

### CODE REVIEW

File: foundation-database/public/tables/itemlocdist.sql [hunk 1]
```diff
@@ -1,5 +1,25 @@
...
+       xt.add_column('itemlocdist', 'itemlocdist_expiration',     'DATE',          NULL, 'public'),
+       xt.add_column('itemlocdist', 'itemlocdist_distlotserial',  'BOOLEAN',       NULL, 'public'),
+       xt.add_column('itemlocdist', 'itemlocdist_warranty',       'DATE',          NULL, 'public'),
+       xt.add_column('itemlocdist', 'itemlocdist_ls_id',          'INTEGER',       NULL, 'public'),
+       xt.add_column('itemlocdist', 'itemlocdist_order_type',     'TEXT',          NULL, 'public'),
+       xt.add_column('itemlocdist', 'itemlocdist_order_id',       'INTEGER',       NULL, 'public'),
+       xt.add_column('itemlocdist', 'itemlocdist_child_series',   'INTEGER',       NULL, 'public'),
+       xt.add_column('itemlocdist', 'itemlocdist_transtype',      'TEXT',          NULL, 'public');
+
+--SELECT xt.add_constraint('itemlocdist', 'itemlocdist_itemlocdist_id_fkey', 'FOREIGN KEY (itemlocdist_itemlocdist_id) REFERENCES itemlocdist(itemlocdist_id)', 'public');
```
Comment (jcarlin) [inline foundation-database/public/tables/itemlocdist.sql]: This is resolved in https://github.com/xtuple/xtuple/pull/3006.

File: foundation-database/public/indexes/itemlocdist.sql [hunk 2]
```diff
@@ -0,0 +1 @@
+select xt.add_index('itemlocdist', 'itemlocdist_source_id', 'itemlocdist_source_id_idx', 'btree', 'public');
```
Comment (jcarlin) [inline foundation-database/public/indexes/itemlocdist.sql]: `itemlocdist_source_id` can be an `itemlocdist_id` or a `location_id`, making it possible that there could be duplicate values in `itemlocdist_source_id`. In fact, even with the case of `itemloc_source_id` being a `itemlocdist_id` value, there could be duplicate values for this column. This index would throw an error in either case I believe.

Comment (gilmoskowitz) [inline foundation-database/public/indexes/itemlocdist.sql]: There's no `UNIQUE` constraint on the index so both cases should be fine.

**Comment by xtuple-ci-user** [comment]
Build finished.

File: foundation-database/public/functions/formatlocationname.sql [hunk 3]
```diff
@@ -1,42 +1,19 @@
...
-
-    IF (_r.location_name IS NOT NULL) THEN
-      _name := (_name || _r.location_name);
-    END IF;
-
-    RETURN _name;
-  ELSE
-    RETURN ''N/A'';
-  END IF;
+  RETURN COALESCE(_name, 'NA');
```
Comment (ChristopherCotnoir) [inline foundation-database/public/functions/formatlocationname.sql]: Should NA still be N/A?

**Pull request merged**